dashboard/app/lib/javascripts/dashboard/main.js

```diff
@@ -65,6 +60,90 @@ window.Dashboard = {
 		Marbles.history.navigate('login'+ redirectPath);
 	},
 
+	__catchInsecurePingResponse: function(httpsArgs) {
+		var httpsXhr = httpsArgs[1], self = this,
+			handleSuccess, handleError;
+
+		handleSuccess = function (httpArgs) {
+			var httpXhr = httpArgs[1];
+			// https did not work but http did...something is wrong with the cert
+			self.Dispatcher.handleAppEvent({
+				name: "HTTPS_CERT_MISSING",
+				status: httpXhr.status
+			});
+		};
+		handleError = function (httpArgs) {
+			var httpXhr = httpArgs[1];
+			if (httpXhr.status === 0) {
+				// https is failing as well...service is unavailable
+				self.Dispatcher.handleAppEvent({
+					name: "SERVICE_UNAVAILABLE",
+					status: httpXhr.status
+				});
+			}
+			// https did not work but http did without a network error
+			// => missing ssl exception for controller
+			self.Dispatcher.handleAppEvent({
+				name: "HTTPS_CERT_MISSING",
+				status: httpXhr.status
+			});
+		};
+
+		if (httpsXhr.status === 0) {
+			// https is unavailable, let's see if http works
+			self.client.ping("controller", "http").catch(handleError).then(handleSuccess);
+			return;
+		}
+		// We got something else than 0 and it's an error. This results in SERVICE_UNAVAILABLE
```

Comment:
`s/something else than 0 and it's an error/an error code other than 0/` and remove the sentence.